storage/store.go

```diff
@@ -823,7 +824,18 @@ func (s *Store) MergeRange(subsumingRng *Range, updatedEndKey proto.Key, subsume
 		return nil, util.Errorf("cannot remove range %s", err)
 	}
 
-	// TODO(bram): The removed range needs to have all of its metadata removed.
+	var readTxn proto.Transaction
+	found, err := engine.MVCCGetProto(s.engine, subsumedKey, s.Clock().Now(), true, nil, &readTxn)
+	if !found || err != nil {
+		return util.Errorf("failed to resolve %s", subsumedKey)
+	}
+	if found {
+		ms := subsumedRng.GetMVCCStats()
+		err = engine.MVCCDelete(s.engine, &ms, subsumedKey, s.Clock().Now(), &readTxn)
```

Comment:
Metadata is more than just the range descriptor: it's also all the raft state and other stats. Basically, everything that calls `MakeRangeIDKey` in https://github.com/cockroachdb/cockroach/blob/1988fb19aec578dbc42f7398e15fe9c42f887728/keys/keys.go#L62